umake/frameworks/ide.py

```diff
@@ -473,3 +474,115 @@ def post_install(self):
                                                      categories=categories))
         if not self.was_in_arduino_group:
             UI.delayed_display(DisplayMessage(_("You need to logout and login again for your installation to work")))
+            
+class BaseNetBeans(umake.frameworks.baseinstaller.BaseInstaller):
+    """The base for all Netbeans installers."""
+    
+    BASE_URL = "http://download.netbeans.org/netbeans/"
+    executable = "nb/netbeans"
+
+    def __init__(self, category, flavour=""):
+        """ The constructror.
+        @param category The IDE category.
+        @param flavour The Netbeans flavour (plugins bundled).
+        """
+        #add a separator to the string, like -cpp
+        if flavour != "":
+            flavour = '-' + flavour
+        self.flavour = flavour
+        
+        super().__init__(name="Netbeans",
+                         description=_("Netbeans IDE"),
+                         category=category, 
+                         only_on_archs=['i386', 'amd64'], #FIXME add arm support
+                         download_page=None,
+                         dir_to_decompress_in_tarball="*", #FIXME windows and mac files not required
```

Comment:
Are all linux/windows/mac stuff in different directories? In that case, you can use something like:
dir_to_decompress_in_tarball="linux-*", (wildcards are supported as you saw)

And the decompressor will do it for you.